extensions/amp-smartlinks/validator-amp-smartlinks.protoascii

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-smartlinks
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-smartlinks"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+
+tags: {  # <amp-smartlinks>
+  html_format: AMP
+  tag_name: "AMP-SMARTLINKS"
+  requires_extension: "amp-smartlinks"
+  attrs: {
+    name: "exclusive-links"
+    value: ""
+  }
+  attrs: {
+    name: "link-attribute"
+  }
+  attrs: {
+    name: "link-selector"
+  }
+  attrs: {
+    name: "linkmate"
+    mandatory: true
+    value: "false"
+    value: "true"
```

Comment:
Rather than `linkmate=true` / `linkmate=false`, have you considered instead using the presence of the attribute as the true/false indication. This would look like:

```
attrs: {
  name: "linkmate"
  value: ""
}
```

This requires a null value, but doesn't mandate the attribute presence. This is a more commonly seen pattern in HTML. Think `<button>` / `<button disabled>` rather than `<button disabled=false>` / `<button disabled=true>`.

Reply:
I can see that -- I think this will likely change in the future, as we move to consolidate functionality with a couple similar extensions.  Would you be open to merging as-is?  Most of our customers are going to take our auto-generated code anyways, and we'd really like to get this merged.

Reply:
We prefer to avoid making something valid for only a short time then having to go through a lengthy deprecation process to make it invalid. It would be better if it can be valid from the start.

Reply:
@c-nichols even if we are going to move to the more generic approach, we will need to keep the existing amp-smartlinks extension for backward compatibility. Do you see any plan to support linkmate value other than `true/false` in the future.  

@Gregable and @honeybadgerdontcare I personally think the current validator spec is the stricter version. We can add `value: ""` on top of `value: "true"` and `value: "false"` without breaking existing page. WDYT